calculator.rb

```diff
@@ -0,0 +1,97 @@
+puts "Thank you for choosing my calculator"
+answer = 0
+
+def is_numeric(s)
+  # This begin / rescue block is for exception catching.
+  # We need it because we're trying to convert string to float and it might fail if string contains non-numeric characters
+  # on successful conversion - it will return true, else false
+  begin
+    Float(s)
+  rescue
+    false # not numeric
+  else
+    true # numeric
+  end
+end
+
+
+def get_numeric_input
+  while user_input = gets.chomp # loop while getting user input
+    isInputOK = is_numeric(user_input)
+    case isInputOK
+    when true
+      break # make sure to break so you don't ask again
+    else
```

Comment:
Since `isInputOK` is a boolean, I think an `if` statement would be more readable that a `case/while` here:

```ruby
if isInputOK
  break # make sure to break so you don't ask again
else
  puts "Please enter a valid number"
end
```